pkg/validate/validate.go

```diff
@@ -189,3 +196,115 @@ func copyAndUnzip(repoFs billy.Filesystem, upstreamPath, localPath string) error
 	}
 	return nil
 }
+
+func CheckImageRancherPrefix() error {
+	index, err := repo.LoadIndexFile("index.yaml")
+	if err != nil {
+		return err
+	}
+	for _, versions := range index.Entries {
+		latestVersion := versions[0]
+		tgzPath := filepath.Join("", latestVersion.URLs[0])
+
+		versionValues, err := DecodeValuesFilesInTgz(tgzPath)
+		if err != nil {
+			logrus.Info(err)
+			continue
+		}
+		chartNameAndVersion := fmt.Sprintf("%s:%s", latestVersion.Name, latestVersion.Version)
+		for _, values := range versionValues {
+			chartImages := PickImagesFromValuesMap(values, chartNameAndVersion)
+			for _, chartImage := range chartImages {
+				if !imageHasRancherPrefix(chartImage) {
+					return fmt.Errorf("Image [%s] does not start with rancher/\n", chartImage)
+				}
+			}
+		}
+	}
+	return nil
+}
```

Comment:
Are all functions in validate.go except `CheckImageRancherPrefix` and `imageHasRancherPrefix` duplicates from the image pkg in rancher/rancher? Just wondering how much was duplicated.

Reply:
Nah they were already modified in the previous version of this PR and I modified them more now with the `list-asset-images` functionality. What might happen is that the way we get images from charts changes, and after changing it in Rancher, we run into it here as well or something but I think that's not a big issue given that the other side is changing Rancher's side and having a dependency there.